filebeat/beater/filebeat.go

```diff
@@ -108,22 +113,42 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	defer func() {
 		// With harvesters being stopped, optionally wait for all enqueued events being
 		// published and written by registrar before continuing shutdown.
-		fb.sigWait.Wait()
+		if !*once {
```

Comment:
no need to check `*once` flag here. Wait will return immediately if no signalers have been added.